docs/docs/workflows/terraform-module.md

```diff
@@ -0,0 +1,152 @@
+# Use the Terraform module
+You can manage a Constellation cluster through Terraform.
+The module package is available as part of the [GitHub release](https://github.com/edgelesssys/constellation/releases/). It consists of a convenience module for each cloud service provider (`{csp}-constellation`) that combines the IAM (`infrastructure/{csp}/iam`), infrastructure (`infrastructure/{csp}`), and constellation (`constellation-cluster`) modules.
+
+## Prerequisites
+- a Linux / Mac operating system
+- a Terraform installation of version `v1.4.4` or above
+
+## Quick setup
+The convenience module allows setting up a Constellation cluster with a single module. It's easiest to consume the module through a remote source, as shown below.
+This allows to upgrade the cluster to a newer Constellation version by simply updating the module source.
+
+:::caution
+
+The current module still relies on the Constellation CLI. Consequently, `terraform apply` creates files such as `constellation.conf.yaml`, `constellation-state.yaml` , `constellation-admin.conf`, `constellation-mastersecret.json` and a directory `constellation-terraform"` containing backups. Make sure to check in these files in your version control when using GitOps.
```

Comment:
Do all of these files have to be kept?

Reply:
No, but most of them. `constellation.conf.yaml` could be reconstructed, but the rest contains cluster information from the apply command.